run-platform.sh

```diff
@@ -298,6 +301,18 @@ run_services() {
   popd 1>/dev/null
 }
 
+show_migration_warning() {
+  # TODO: Avoid showing warning if migration is done / not required
+  echo -e "\n########################## IMPORTANT NOTICE ##########################"
+  echo -e "From ${yellow_text}$V2_WARNING_VERSION${default_text} onwards, a ${red_text}data migration${default_text} is required to"
+  echo -e "continue using the platform with your existing data."
+  echo -e "For detailed instructions on how to perform this migration, "
+  echo -e "please check the following guide:"
+  echo -e "    ${blue_text}backend/migrating/v2/README.md${default_text}"
+  echo -e "\nPlease ignore this warning if you've migrated already."
+  echo -e "#######################################################################"
+}
+
```

Comment:
@chandrasekharan-zipstack We could generalize this for all future versions.

1. Rename function to `show_release_warnings` to make it generic.
2. Keep adding release notes for any release for which we need to show a warning.
3. If `current_version` < `warning_version` and `target_version` >= `warning_version`, then show the warnings for all the release versions up to target version.
4. We could have a sub folder inside `scripts` where we maintain the logic above and also a JSON containing warning versions and their corresponding messages.
5. `run-platform.sh` could invoke above script. In this way, we could abstract away all version comparison logic required to show warnings.

Reply:
@hari-kuriakose regarding point 3
> If current_version < warning_version and target_version >= warning_version, then show the warnings for all the release versions up to target version.

- Can you elaborate on these 3 - `current_version`, `warning_version` and `target_version`?
- Do you mean a logical AND `current_version < warning_version and target_version >= warning_version`
here?

Reply:
@hari-kuriakose its a bit tricky to correctly get the current version reliably - to not remain blocked, I've added the script to print warnings (check screenshot) and commented out the call from `./run-platform.sh` for the time being. We can revisit this when we need to update the v2 migration version in the new `warnings.json` / `README.md`

I prefer not blocking this PR further for the generic functionality of printing warnings

cc: @muhammad-ali-e

Reply:
Resolved.